src/account.ts

```diff
@@ -310,12 +310,18 @@ export class Account {
      * @returns {Promise<FinalExecutionOutcome>}
      * TODO: expand this API to support more options.
      */
-    async addKey(publicKey: string | PublicKey, contractId?: string, methodName?: string, amount?: BN): Promise<FinalExecutionOutcome> {
+    async addKey(publicKey: string | PublicKey, contractId?: string, methodNames?: string|string[], amount?: BN): Promise<FinalExecutionOutcome> {
+        if (!methodNames) {
+            methodNames = []
```

Comment:
@mattlockyer are you sure this is the same as `null` / `undefined`?  I think empty list might mean "no methods are allowed" and `null` means "any method is allowed".

Reply:
In the core:
Empty list means "all methods" at protocol level.
Empty string `''` means no methods.
https://nomicon.io/DataStructures/AccessKey.html#accesskeypermissionfunctioncall

It seems like we have options how to interpret this?

AFAIK we have to end up with an Array in TS/JS?
https://github.com/near/near-api-js/blob/cf003f1cadb5bfa93dcde165f94dfadd91a9f82e/src/transaction.ts#L9-L13